Sources/SPMBuildCore/Sanitizers.swift

```diff
@@ -12,12 +12,31 @@ import TSCBasic
 import TSCUtility
 
 /// Available runtime sanitizers.
-public enum Sanitizer: String, Encodable {
+public enum Sanitizer: String, Encodable, CaseIterable {
     case address
     case thread
     case undefined
     case scudo
 
+    public init(argument: String) throws {
+        if let sanitizer = Sanitizer(rawValue: argument) {
+            self = sanitizer
+            return
+        }
+
+        for sanitizer in Sanitizer.allCases where sanitizer.shortName == argument {
+            self = sanitizer
+            return
+        }
+
+        throw ArgumentConversionError.custom("valid sanitizers: \(Sanitizer.formattedValues)")
+    }
+
+    /// All sanitizer options in a comma separated string
+    public static var formattedValues: String {
+        return Sanitizer.allCases.map(\.rawValue).joined(separator: ", ")
+    }
+
```

Comment:
These seem like concerns that should be internal to the command-line interface, rather than additions to the public API. What do you think about moving these into a `fileprivate` extension alongside `SwiftToolOptions`?

Reply:
I've moved it, but note the extension has to be `public` to make this initializer a valid implementing of `StringEnumArgument`